diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f708e75..32cf71d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -269,6 +269,7 @@ if(MI_BUILD_SHARED)
     add_custom_command(TARGET mimalloc POST_BUILD
       COMMAND "${CMAKE_COMMAND}" -E copy "${CMAKE_CURRENT_SOURCE_DIR}/bin/mimalloc-redirect.dll" $<TARGET_FILE_DIR:mimalloc>
       COMMENT "Copy mimalloc-redirect.dll to output directory")
+    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/bin/mimalloc-redirect.dll DESTINATION ${mi_install_dir}/bin)
   endif()
 
   if (MI_INSTALL_TOPLEVEL MATCHES "ON")
@@ -279,7 +280,7 @@ if(MI_BUILD_SHARED)
   else()
     install(TARGETS mimalloc EXPORT mimalloc DESTINATION ${mi_install_dir} LIBRARY)
   endif()
-  install(EXPORT mimalloc DESTINATION ${mi_install_dir}/cmake)
+  install(EXPORT mimalloc DESTINATION ${mi_install_dir}/lib/cmake/mimalloc)
 endif()
 
 # static library
@@ -309,8 +310,8 @@ endif()
 install(FILES include/mimalloc.h DESTINATION ${mi_install_dir}/include)
 install(FILES include/mimalloc-override.h DESTINATION ${mi_install_dir}/include)
 install(FILES include/mimalloc-new-delete.h DESTINATION ${mi_install_dir}/include)
-install(FILES cmake/mimalloc-config.cmake DESTINATION ${mi_install_dir}/cmake)
-install(FILES cmake/mimalloc-config-version.cmake DESTINATION ${mi_install_dir}/cmake)
+install(FILES cmake/mimalloc-config.cmake DESTINATION ${mi_install_dir}/lib/cmake/mimalloc)
+install(FILES cmake/mimalloc-config-version.cmake DESTINATION ${mi_install_dir}/lib/cmake/mimalloc)
 
 if(NOT WIN32 AND MI_BUILD_SHARED AND NOT MI_INSTALL_TOPLEVEL)
   # install a symlink in the /usr/local/lib to the versioned library
