diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b56953c..2b29068 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -208,10 +208,12 @@ endif()
 # -----------------------------------------------------------------------------
 
 if (MI_INSTALL_TOPLEVEL)
+  set(mi_install_bindir   "bin")
   set(mi_install_libdir   "lib")
   set(mi_install_incdir   "include")
   set(mi_install_cmakedir "cmake")
 else()
+  set(mi_install_libdir   "bin/mimalloc-${mi_version}")
   set(mi_install_libdir   "lib/mimalloc-${mi_version}")
   set(mi_install_incdir   "include/mimalloc-${mi_version}")
   set(mi_install_cmakedir "share/mimalloc-${mi_version}/cmake") 
@@ -279,9 +281,14 @@ if(MI_BUILD_SHARED)
     add_custom_command(TARGET mimalloc POST_BUILD
       COMMAND "${CMAKE_COMMAND}" -E copy "${CMAKE_CURRENT_SOURCE_DIR}/bin/mimalloc-redirect${MIMALLOC_REDIRECT_SUFFIX}.dll" $<TARGET_FILE_DIR:mimalloc>
       COMMENT "Copy mimalloc-redirect${MIMALLOC_REDIRECT_SUFFIX}.dll to output directory")
+    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/bin/mimalloc-redirect${MIMALLOC_REDIRECT_SUFFIX}.dll DESTINATION ${mi_install_bindir})
   endif()
 
-  install(TARGETS mimalloc EXPORT mimalloc DESTINATION ${mi_install_libdir} LIBRARY)  
+  install(TARGETS mimalloc EXPORT mimalloc LIBRARY
+    RUNTIME DESTINATION ${mi_install_bindir}
+    LIBRARY DESTINATION ${mi_install_libdir}
+    ARCHIVE DESTINATION ${mi_install_libdir}
+  )  
   install(EXPORT mimalloc DESTINATION ${mi_install_cmakedir})
 endif()
 
